pydra/engine/tests/test_workflow.py

```diff
@@ -2217,13 +2251,11 @@ def test_wfasnd_ndst_updatespl_1(plugin, tmpdir):
     """
     wfnd = Workflow(name="wfnd", input_spec=["x"])
     wfnd.add(add2(name="add2", x=wfnd.lzin.x))
+    wfnd.add2.split("x", x=[2, 4])
     wfnd.set_output([("out", wfnd.add2.lzout.out)])
-    # TODO: without this the test is failing
-    wfnd.inputs.x = [2, 4]
 
     wf = Workflow(name="wf", input_spec=["x"])
     wf.add(wfnd)
-    wfnd.add2.split("x")
     wf.set_output([("out", wf.wfnd.lzout.out)])
```

Comment:
This won't work because, you need to provide a value for x in the line

```python
existing_wf.add2.split('x')
```

I think it might also fail if any of the fields were typed and connected to downstream nodes in the existing_wf as they would be expecting a single value not a state-array.

However, I'm not sure of the use case for this test. When would you want to go into the inner workflow and split only one node and not an input?

If your inner workflow has A->B->C, then you wouldn't be able to split either B or C as the inputs coming from A and B won't be lists. Even if it is on an input which is connected to both A & B, you wouldn't be able to provide a list to one and a single value to the other. So I can't think of a case where would work unless your tasks have inputs of type `Union[list[T], T]`.

Reply:
I'm not sure if we ever had very specific example for it... Just wanted to be possible, but probably you're right that there are not many use cases...